File: app/src/main/java/unilife/com/unilife/profile/adapters/InterestAdapter.java
package unilife.com.unilife.profile.adapters;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import java.util.ArrayList;

import butterknife.BindView;
import butterknife.ButterKnife;
import unilife.com.unilife.R;
import unilife.com.unilife.profile.model.UserInterest;
import unilife.com.unilife.profile.model.UserLanguage;

public class InterestAdapter extends RecyclerView.Adapter<InterestAdapter.MyViewHolder> {

    ArrayList<UserInterest> arrayList = new ArrayList<>();
    Context context;

    // Provide a suitable constructor (depends on the kind of dataset)
    public InterestAdapter(Context context) {
        this.context = context;
    }

    public void updateData(ArrayList<UserInterest> arrayList) {
        this.arrayList = arrayList;
        notifyDataSetChanged();
    }

    // Create new views (invoked by the layout manager)
    @Override
    public InterestAdapter.MyViewHolder onCreateViewHolder(ViewGroup parent,
                                                           int viewType) {
        // create a new view
        View v = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.adapter_profile_2, parent, false);

        MyViewHolder vh = new MyViewHolder(v);
        return vh;
    }

    // Replace the contents of a view (invoked by the layout manager)
    @Override
    public void onBindViewHolder(MyViewHolder holder, int position) {
        // - get element from your dataset at this position
        // - replace the contents of the view with that element
        holder.textView.setText(arrayList.get(position).getInterestName());
    }

    // Return the size of your dataset (invoked by the layout manager)
    @Override
    public int getItemCount() {
        return arrayList.size();
//        return arrayList.size();
    }

    // Provide a reference to the views for each data item
    // Complex data items may need more than one view per item, and
    // you provide access to all the views for a data item in a view holder
    public static class MyViewHolder extends RecyclerView.ViewHolder {
        // each data item is just a string in this case

        @BindView(R.id.text)
        TextView textView;

        public MyViewHolder(View v) {
            super(v);
            ButterKnife.bind(this, v);

        }
    }
}

File: app/src/main/java/unilife/com/unilife/profile/InterestAdapter.kt
package unilife.com.unilife.profile

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import com.squareup.picasso.Picasso
import unilife.com.unilife.R
import unilife.com.unilife.retrofit.WebUrls

class InterestAdapter() : RecyclerView.Adapter<InterestAdapter.InterestViewHolder>() {

    lateinit var context: Context
    lateinit var interestdata:ArrayList<InterestModel>

    constructor(context :Context,interestdata: ArrayList<InterestModel>) : this(){

        this.context=context
        this.interestdata=interestdata

    }



    override fun onCreateViewHolder(p0: ViewGroup, p1: Int): InterestViewHolder {
        val itemView = LayoutInflater.from(p0.context).inflate(R.layout.hobbies_itemlist, p0, false)
        return InterestViewHolder(itemView)
    }

    override fun getItemCount(): Int {
        return interestdata.size
    }

    override fun onBindViewHolder(interestViewHolder: InterestViewHolder, p1: Int) {
        /*interestViewHolder.interest_title?.visibility=View.INVISIBLE*/
        Picasso.with(context).load(WebUrls.MYPROFILE_ICON+interestdata[p1].interesticon)
            .placeholder(R.drawable.no_image).into(interestViewHolder.interest_icon)
        interestViewHolder.interest_title?.text = interestdata[p1].interestname

    }


    inner class InterestViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        internal var interest_icon: ImageView? = null
        internal var interest_title: TextView? = null
        init {
            interest_icon=itemView.findViewById(R.id.hobby_icon)
            interest_title=itemView.findViewById(R.id.tvhobbies_itemlist)
        }
    }

}
